pkg/kubelet/volumes.go

```diff
@@ -77,12 +77,18 @@ func (kl *Kubelet) mountExternalVolumes(pod *api.BoundPod) (volumeMap, error) {
 	for i := range pod.Spec.Volumes {
 		volSpec := &pod.Spec.Volumes[i]
 
+		podRef, err := api.GetReference(pod)
```

Comment:
I'm wondering now if allowing an empty SelfLink for a BoundPod is correct or if, perhaps, I found a bug.

It seems like the BoundPod *should* have a SelfLink:

https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/registry/pod/bound_pod_factory.go#L36